src/stores/my-account.ts

```diff
@@ -13,50 +14,73 @@ type State = {
   address: string | null
   signer: Signer | null
   energy: number | null
-  unsubscribeEnergy: () => void
   encodedSecretKey: string | null
+  _unsubscribeEnergy: () => void
+  _currentSessionSecretKey: string
+  _isNewSessionKey: boolean
+}
+
+type Session = {
+  encodedSecretKey: string
+  address: string
+  isNewSecretKey?: boolean
 }
 type Actions = {
-  login: (secretKey: string, isInitialization?: boolean) => Promise<boolean>
+  login: (
+    secretKey?: string,
+    isInitialization?: boolean
+  ) => Promise<string | false>
   logout: () => void
-  subscribeEnergy: () => Promise<void>
+  _subscribeEnergy: () => Promise<void>
+  _setSessionKey: (session: Session) => Promise<void>
+  _syncSessionKey: () => Promise<void>
+  _syncSessionWithLocalStorage: () => Promise<void>
+  _getSecretKeyForLogin: () => Promise<string>
 }
 
-const STORAGE_KEY = 'account'
+const ACCOUNT_STORAGE_KEY = 'account'
+const SESSION_STORAGE_KEY = 'session'
 
-const initialState = {
+const initialState: State = {
   isInitializedAddress: true,
   address: null,
   signer: null,
   energy: null,
-  unsubscribeEnergy: () => undefined,
   encodedSecretKey: null,
+  _currentSessionSecretKey: '',
```

Comment:
You use `_` before name for private methods, right?

Reply:
yes

Reply:
Can I get it from other components?